core/lib/rom/schema/dsl.rb

```diff
@@ -72,7 +75,7 @@ def attribute(name, type, options = EMPTY_HASH)
                          "Attribute #{ name.inspect } already defined"
         end
 
-        attributes[name] = build_type(name, type, options)
+        attributes[name] = build_attribute_info(name, type, options)
```

Comment:
In order to be backward compatible with plugins redefining `#build_type`, I take an intermediate step here which forwards to `#build_type` and takes the options.